### PR TITLE
feat: extend context and contextkey service

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

This PR extends the context/contextkey implemention.
The context is now able to retrieve values using complex keys such as 'key.prop1.prop2'. This way if the context value is an object we can get a property value.

The contextKey is extended to support and, or, equal, not equal, true, false, Lt, LtEq, Gt, GtEq, EOF, RegexOp and RegexStr operations.

All of this will be used in the onboarding PR.

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

N/A, this is part of the onboarding PR

### How to test this PR?

run tests

### CODE REVIEW

File: packages/renderer/src/lib/context/scanner.ts [hunk 1]
```diff
@@ -26,8 +26,13 @@ import { CharCode } from '../../../../main/src/plugin/util/charCode';
 export const enum TokenType {
   LParen,
   RParen,
+  Eq,
```
Comment (benoitf) [inline packages/renderer/src/lib/context/scanner.ts:30]: is there a reason why you removed these options from the code we imported ?

like NotEq is not here, as well Lt, LtEq, Gt, GtEq, EOF and RegexOp and RegexStr ?

Comment (lstocchi) [reply packages/renderer/src/lib/context/scanner.ts:30]: i'm just adding the support for the rules i actually use

Comment (benoitf) [reply packages/renderer/src/lib/context/scanner.ts:30]: ok but why not let them here as they were already implemented ?

Comment (lstocchi) [reply packages/renderer/src/lib/context/scanner.ts:30]: ok i'll add everything

**Review by benoitf — APPROVED** [review]

**Comment by lstocchi** [comment]
@benoitf not sure if the approve is still valid after the new commit. Can you please re-approve it after you checked, so i'm sure.
Then can you remind me how to mark the sonar issues as "won't fix"? They are all related to the vscode code

**Comment by benoitf** [comment]
@lstocchi I think we need to solve the issue if possible as Sonar is popping out again the issue so it's not easy to mark it as ignore permanently ..

**Review by benoitf — APPROVED** [review]

**Review by cdrage — APPROVED** [review]

**Pull request merged**